src/jdns/jdns_sys.c

```diff
@@ -271,8 +271,8 @@ static jdns_string_t *file_nextline(FILE *f)
 	at = 0;
 	while(1)
 	{
-		unsigned char c = fgetc(f);
-		if(feof(f))
+		unsigned char c = (unsigned char)fgetc(f); // cast is intentional
+		if(feof(f)) // checking EOF here
```

Comment:
honestly if one knows how fgetc works then it's clear what will be assigned to c.
and if one does not know fgetc then your comments will say nothing.

Instead I'd add a comment similar to what you just said
// can return EOF(-1),  check it explicitly on the next line